src/DecFP.jl

```
@@ -498,15 +498,36 @@ end # widths w
 
 Base.round(x::DecimalFloatingPoint, ::RoundingMode{:FromZero}) = signbit(x) ? floor(x) : ceil(x)
 
-Base.trunc(::Type{Integer}, x::DecimalFloatingPoint) = trunc(Int, x)
-Base.floor(::Type{Integer}, x::DecimalFloatingPoint) = floor(Int, x)
-Base.ceil(::Type{Integer}, x::DecimalFloatingPoint) = ceil(Int, x)
-Base.round(::Type{Integer}, x::DecimalFloatingPoint) = round(Int, x)
-Base.round(::Type{Integer}, x::DecimalFloatingPoint, ::RoundingMode{:NearestTiesAway}) = round(Int, x, RoundNearestTiesAway)
-Base.convert(::Type{Integer}, x::DecimalFloatingPoint) = convert(Int, x)
+for (f) in (:trunc, :floor, :ceil)
+    @eval function Base.$f(::Type{I}, x::DecimalFloatingPoint) where {I<:Integer}
+        I′ = I
```

Maybe
```jl
I′ = isabstracttype(I) ? (I <: Unsigned ? UInt : I) : I
```

---

No, that's not right either, since `Int` is not necessarily a subtype of an arbitrary abstract `Integer` subtype.

Seems like it would be better just to use `I` (no `I′`, and let it throw a `MethodError` for `typemin` and `typemax` if `I` i s an abstract type that doesn't support those.